packages/Discretization/test/tstMeshGenerator.cpp

```diff
@@ -0,0 +1,775 @@
+/****************************************************************************
+ * Copyright (c) 2012-2017 by the DataTransferKit authors                   *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the DataTransferKit library. DataTransferKit is     *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ ****************************************************************************/
+
+#include "MeshGenerator.hpp"
+#include <Teuchos_UnitTestHarness.hpp>
+
```

Comment:
There is only so much you can test here. I am not so keen on pretty much mirroring the actual implementation and asserting the connectivity and the coordinates are the same...

Reply:
I have changed the structured and mixed tests to read input files. I didn't change the simplex because the code is a lot easier to read than the input file in this case. I also want to point out that the way the meshes are implemented in the tests is significantly different than what is done in the mesh generator. If any thing the input files approach is closer to the mesh generator implementation.